datafusion/sqllogictest/test_files/scalar.slt

```diff
@@ -1878,3 +1878,51 @@ query T
 SELECT CONCAT('Hello', 'World')
 ----
 HelloWorld
+
+statement ok
+CREATE TABLE simple_string(
+  letter STRING,
+  letter2 STRING
+) as VALUES
+  ('A', 'APACHE'),
+  ('B', 'APACHE'),
+  ('C', 'APACHE'),
+  ('D', 'APACHE')
+;
+
+query TT
+EXPLAIN SELECT letter, letter = LEFT('APACHE', 1) FROM simple_string;
+----
+logical_plan
+Projection: simple_string.letter, simple_string.letter = Utf8("A") AS simple_string.letter = left(Utf8("APACHE"),Int64(1))
```

Comment:
The end-to-end tests are not failed, but I think it is good to have it to be clear.

For end-to-end execution, `LEFT('APACHE', 1)` is a foldable literal. So expression optimizer optimizes it to a scalar literal.

But if developers use the API `create_physical_expr` directly, the reported error will happen.